runtime/src/launch/mpirun4ofi/launch-mpirun4ofi.c

```diff
@@ -34,17 +35,32 @@ static char** chpl_launch_create_argv(const char *launch_cmd,
   static char _nlbuf[16];
   snprintf(_nlbuf, sizeof(_nlbuf), "%d", getArgNumLocales());
 
-  char* largv[] = { (char*) launch_cmd,
-                    (char*) "-np",
-                    (char*) _nlbuf,
-                    (char*) "-map-by",
-                    (char*) "ppr:1:node",
-                    (char*) "-map-by",
-                    (char*) "node:oversubscribe",
-                    (char*) "-bind-to",
-                    (char*) "none",
-                  };
-  const int largc = sizeof(largv) / sizeof(largv[0]);
+  char* largv[20];
+  int largc = 0;
+#define ADD_LARGV(s)                                                    \
+  do {                                                                  \
+    if (largc < sizeof(largv) / sizeof(largv[0])) {                     \
+      largv[largc++] = (char*) (s);                                     \
+    } else {                                                            \
+      chpl_internal_error("too many mpirun command line arguments");    \
```

Comment:
Seems like this is asking for trouble. Can we just dynamically allocate largv and realloc here?

Reply:
We're not supposed to spend any time on launchers.   And presumably especially not stopgap ones.

Reply:
I don't think we're not supposed to spend any time on launchers, just not significant amounts of time or do major refactoring/improvements.  I don't think this would take long (but could be wrong) and so far as I know we don't really have a near term plan for replacing this launcher with something better.